lms/views/api/canvas/authorize.py

```diff
@@ -84,7 +96,7 @@ def oauth2_redirect_error(self):
         template_variables = {
             "invalid_scope": self.request.params.get("error") == "invalid_scope",
             "details": self.request.params.get("error_description"),
-            "scopes": self.scopes,
+            "scopes": self.files_scopes + self.sections_scopes,
```

Comment:
This is the list of scopes that we pass to the "You have missing scopes" error page's template. We always list both the files and sections scopes on this error page, even if sections is disabled, because:

1. In this exception view you can't always tell whether sections is enabled or not. Depending on exactly what happened `request.lti_user` is sometimes `None` when this view is called and you don't have the consumer key and can't look up the application instance.

2. We want people to give us all the scopes anyway. Even if they aren't using the sections feature right now, it'll be easier for them if/when they decide to ask us to enable sections for them if they already have the sections scopes. There's no advantage in _not_ asking people to set the sections scopes, this will only create more work for us and them.